### PR TITLE
Fix buffer_wrapper test for C++ 11 and 14

### PR DESCRIPTION
**Pull request by dparanic**

### CODE REVIEW

File: test/general/buffer_wrapper.pass.cpp [hunk 1]
```diff
@@ -34,7 +34,8 @@ struct test_buffer_wrapper
         EXPECT_TRUE(end - begin == size, "wrong effect of iterator's operator - iterator");
 
         auto buf = begin.get_buffer();
-        EXPECT_TRUE(sycl::host_accessor(buf, sycl::read_only).get_pointer() == data, "wrong effect of iterator's method get_buffer");
+        T* actual_data = sycl::host_accessor<uint32_t, 1, sycl::access_mode::read>(buf).get_pointer();
```
Comment (capatober) [inline test/general/buffer_wrapper.pass.cpp:37]: Could you please try to use `get_host_access` method? I believe it can reduce number of arguments

Comment (dparanic) [reply test/general/buffer_wrapper.pass.cpp:37]: Method `get_host_access` is available for C++ 17 or higher. Should we use it still? Should we use macro for it?

Comment (capatober) [reply test/general/buffer_wrapper.pass.cpp:37]: It would be better if we understood why it's not available. However, this is not a blocker

Comment (capatober) [inline test/general/buffer_wrapper.pass.cpp:37]: It's better to use `T` instead of hardcoded `uint32_t`

Comment (dparanic) [reply test/general/buffer_wrapper.pass.cpp]: Let's keep it unchanged. method `get_host_access` requires CTAD support (Class template argument deduction), that's why it is available for C++ 17 or higher.

Comment (dparanic) [reply test/general/buffer_wrapper.pass.cpp]: Done

**Review by capatober — APPROVED** [review]

**Pull request merged**